qiskit_experiments/calibration_management/base_calibration_experiment.py

```diff
@@ -0,0 +1,361 @@
+           method.
+
+        #. Use any default schedules specified by the :meth:`get_schedules_from_defaults`.
+
+        If any one step does not return a schedule then we attempt to get schedules from the next
+        step. If none of these three steps have returned any schedules then an error is raised.
+
+        Args:
+            qubits: The qubits for which to get the schedule in the calibrations. If None is given
+                this will default to the physical qubits of the experiment.
+            sched_name: The name of the schedule to retrieve from the instance of
+                :class:`Calibrations` stored as a protected variable. If this is None then
+                :meth:`get_schedule_from_calibrations` will default to the :code:`self._sched_name`.
+            option_name: The name of the option under which to get the schedule from the experiment
+                options. This will default to "schedule" if None is given.
+            assign_params: A dict that :meth:`get_schedule_from_calibrations` can use to leave
+                certain parameters in the schedule unassigned. The key is the name of the parameter
+                and the value should be an instance of :class:`ParameterExpression`.
+            kwargs: Additional keyword arguments that can be used by implementations of
+                :meth:`get_schedule_from_defaults`.
+
+        Returns:
+            schedules: The schedules (possibly with one or more free parameters) as either a
+                ScheduleBlock or a list of ScheduleBlocks depending on the experiment.
+
+        Raises:
+            CalibrationError: if none of the methods above returned schedules.
+        """
+        schedules = self._get_schedule_from_options(option_name)
+
+        if schedules is None:
+            schedules = self._get_schedule_from_calibrations(qubits, sched_name, assign_params)
+
+        if schedules is None:
+            schedules = self._get_schedule_from_defaults(**kwargs)
+
+        self._validate_schedule(schedules)
+
+        return schedules
+
+    def run(
+        self,
+        backend: Backend,
+        analysis: bool = True,
+        experiment_data: Optional[ExperimentData] = None,
+        **run_options,
+    ) -> ExperimentData:
+        """Run an experiment, perform analysis, and update any calibrations.
+
+        Args:
+            backend: The backend to run the experiment on.
+            analysis: If True run analysis on the experiment data.
+            experiment_data: Optional, add results to existing experiment data.
+                If None a new ExperimentData object will be returned.
+            run_options: backend runtime options used for circuit execution.
+
+        Returns:
+            The experiment data object.
+        """
+        experiment_data = super().run(backend, analysis, experiment_data, **run_options)
+
+        if self.auto_update:
+            experiment_data = experiment_data.block_for_results()
```

Comment:
This blocking isn't needed now that #407 is merged. You should be able to do:
```python
if self.auto_update:
    experiment_data.add_analysis_callback(self.updated_calibrations)
```
Which will have the update added to the async job so it runs after the analysis has finished